apps/server/src/modules/school/service/school.service.ts

```diff
@@ -34,14 +35,20 @@ export class SchoolService {
 
 	async setMigration(
 		schoolId: EntityId,
-		oauthMigrationPossible: boolean,
-		oauthMigrationMandatory: boolean,
-		oauthMigrationFinished: boolean
+		oauthMigrationPossible?: boolean,
+		oauthMigrationMandatory?: boolean,
+		oauthMigrationFinished?: boolean
 	): Promise<OauthMigrationDto> {
 		const schoolDo: SchoolDO = await this.schoolRepo.findById(schoolId);
-		schoolDo.oauthMigrationPossible = oauthMigrationPossible ? new Date() : undefined;
-		schoolDo.oauthMigrationMandatory = oauthMigrationMandatory ? new Date() : undefined;
-		schoolDo.oauthMigrationFinished = oauthMigrationFinished ? new Date() : undefined;
+		if (isDefined(oauthMigrationPossible)) {
+			schoolDo.oauthMigrationPossible = oauthMigrationPossible ? new Date() : undefined;
+		}
+		if (isDefined(oauthMigrationMandatory)) {
+			schoolDo.oauthMigrationMandatory = oauthMigrationMandatory ? new Date() : undefined;
+		}
+		if (isDefined(oauthMigrationFinished)) {
+			schoolDo.oauthMigrationFinished = oauthMigrationFinished ? new Date() : undefined;
```

Comment:
It's true that the client only sends one boolean at a time except in the case of oauthMigrationFinished, where the finished timestamp is set AND  the oauthMigrationPossible timestamp is removed. With this logic, the second part would be ignored. Should we fix this behavior here or with the array of services which will come from finishing the migration?

Reply:
All 2 attributes are checked individually. So no matter the combination it will be set correctly